photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/DataProcessingUtils.scala

```diff
@@ -117,9 +117,24 @@ object DataProcessingUtils {
     } else {
       1.0
     }
+
+    val metaMap = Utils.getMapAvro(record, AvroFieldNames.META_DATA_MAP)
```

Comment:
Looks like if ```META_DATA_MAP``` is null in the record, an ```IllegalArgumentException``` will be thrown.

One suggestion is to return an empty ```Map[String, JObject]``` instead of a ```Null```.